src/cio_os.c

```diff
@@ -58,6 +60,12 @@ int cio_os_mkpath(const char *dir, mode_t mode)
         return 0;
     }
 
-    cio_os_mkpath(dirname(strdupa(dir)), mode);
+    dup_dir = strdup(dir);
```

Comment:
You really need to check the return value here.
If it ever fails, it silently breaks the logic in a very nasty way
(because `dirname(NULL)` yields `/`).

Reply:
Added checking return value in: https://github.com/edsiper/chunkio/pull/1/commits/7a4b1255079a53b0a6f05e23fa4b27970b2bcf9b.

Reply:
OK. Thanks for the fix.
BTW, why did you add a line to set `errno` manually in c797d14?
AFAICT `strdup(3)` should take care of it.